Mathlib/MeasureTheory/Measure/Haar/DistribChar.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+/-!
+# The distributive character of Haar measures
+
+Given a group `G` acting on an measurable additive commutative group `A`, and an element `g : G`,
+one can pull back the Haar measure `μ` of `A` along the map `(g • ·) : A → A` to get another Haar
+measure `μ'` on `A`. By unicity of Haar measures, there exists some nonnegative real number `r` such
+that `μ' = r • μ`. We can thus define a map `distribHaarChar : G → ℝ≥0` sending `g` to its
+associated real number `r`. Furthermore, this number doesn't depend on the Haar measure `μ` we
+started with, and `distribHaarChar` is a group homomorphism.
+
+## See also
+
+[Zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/canonical.20norm.20coming.20from.20Haar.20measure/near/480050592)
+-/
+
+open MeasureTheory.Measure
+open scoped NNReal Pointwise ENNReal
+
+namespace MeasureTheory
+variable {G A : Type*} [Group G] [AddCommGroup A] [DistribMulAction G A] [MeasurableSpace A]
+  -- We only need `MeasurableConstSMul G A` but we don't have this class. So we erroneously must
+  -- assume `MeasurableSpace G` + `MeasurableSMul G A`
+  [MeasurableSpace G] [MeasurableSMul G A] [TopologicalSpace A] [BorelSpace A]
+  [IsTopologicalAddGroup A] [LocallyCompactSpace A] [ContinuousConstSMul G A] {μ ν : Measure A}
+  {g : G} [μ.IsAddHaarMeasure]
+
+variable (A) in
+/-- The distributive Haar character of a group `G` acting distributively on a group `A` is the
+unique positive real number `Δ(g)` such that `μ (g • s) = Δ(g) * μ s` for all Haar
+measures `μ : Measure A`, set `s : Set A` and `g : G`. -/
+@[simps -isSimp]
+noncomputable def distribHaarChar : G →* ℝ≥0 where
```

Comment:
Is there a reason to prefer `ℝ≥0` over `ℝ` with a separate proof of nonnegativity?

Reply:
Yes. You can easily turn that into a group hom to the morally correct group, namely `Units NNReal`. I don't see how you would do that for the `Real`-valued version (except by defining the `NNReal`-valuzd version from it).